packages/gatsby/src/schema/schema.js

```diff
@@ -190,11 +191,29 @@ const updateSchemaComposer = async ({
     parentSpan: parentSpan,
   })
   activity.start()
-  await printTypeDefinitions({
-    config: printConfig,
-    schemaComposer,
-    parentSpan: activity.span,
-  })
+  if (!process.env.GATSBY_SKIP_WRITING_SCHEMA_TO_FILE) {
+    await printTypeDefinitions({
+      config: printConfig,
+      schemaComposer,
+      parentSpan: activity.span,
+    })
+    if (_CFLAGS_.GATSBY_MAJOR === `4` && shouldGenerateEngines()) {
+      // make sure to print schema that will be used when bundling graphql-engine
+      const graphqlEngineSnapshotPath = process.cwd() + `/.cache/schema.gql`
```

Comment:
do we want to use process.cwd here? Shouldn't we use `store.getState().program.directory`?

Reply:
I've refactored this whole piece